src/Observations.jl

```diff
@@ -269,11 +269,12 @@ end
 ##### FieldTimeSeriesCollector for collecting data while a simulation runs
 #####
 
-struct FieldTimeSeriesCollector{G, D, F, T}
+struct FieldTimeSeriesCollector{G, D, F}
     grid :: G
+    times :: Vector{Float64}
     field_time_serieses :: D
     collected_fields :: F
-    times :: T
+    collection_times :: Vector{Float64}
```

Comment:
why `T`-> `Vector{Float64}`?

Reply:
Do we need anything else? It's a little simpler.

Reply:
Happy to go back for sure!

But we do need `collection_times` to be a vector since we need to zero it out.

Reply:
Nah, I think we are all good. It might break if somebody does `times = [1, 2, 3]` but that's alright!

Reply:
I think thats ok:

```julia
julia> struct Test
       a::Vector{Float64}
       end

julia> t = Test([1, 2, 3])
Test([1.0, 2.0, 3.0])
```

Reply:
It doens't work for `tuples` (we'd have to call `collect`)

Reply:
ranges are ok too: 

```
julia> t = Test(1:2)
Test([1.0, 2.0])
```

Reply:
better could be

```
times :: T
collection_times :: Vector{Float64}
```

and 

```
collection_times = zeros(size(times)...)
```

We can also relax `Float64`.

Reply:
Great! Leave it as is!